src/hooks/useDeepCompareRef.ts

```diff
@@ -0,0 +1,16 @@
+import { useRef } from 'react'
+import { isDeepEqual } from 'utils'
+
+function useDeepCompareRef(value: undefined): undefined
+function useDeepCompareRef<R>(value: R): R
+function useDeepCompareRef<R>(value?: R): R | undefined {
+  const ref = useRef<R | undefined>()
+
+  if (!isDeepEqual(value, ref.current)) {
```

Comment:
shouldn't this code be wrapped in a useEffect? or since there's only on e parameter doesn't matter?

Reply:
no its not really a hook, it's using `useRef` and the `use` prepend just tells react that it should follow `hook` rules e.g not being callled outside the function or being called conditionally, etc

https://reactjs.org/docs/hooks-custom.html#using-a-custom-hook

Reply:
what I mean is `isDeepEqual` would be executed on every re-render of `useDeepCompareRef` that if `value` and `ref.current` don't change (the reference), you know you don't need to update the `ref.current = value` therefore you also save some calls to `isDeepEqual`
--> probably is not too important if the object has not too much deph